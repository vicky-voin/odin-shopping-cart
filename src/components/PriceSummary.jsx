import styles from "./PriceSummary.module.css";
import formatPrice from "../utils/priceConverter";

export default function PriceSummary({ subtotal }) {
  const shipping = subtotal === 0 ? 0 : 15; //hardcoded shipping value for demo purposes
  const tax = 0.13 * subtotal; //hardcoded 13% tax for demo purposes
  const total = subtotal + shipping + tax;

  return (
    <div>
      <div>Subtotal</div>
      <div data-testid="subtotalValue">{formatPrice(subtotal)}</div>
      <div>Shipping</div>
      <div data-testid="shippingValue">{formatPrice(shipping)}</div>
      <div>Tax</div>
      <div data-testid="taxValue">{formatPrice(tax)}</div>
      <div>Order total</div>
      <div data-testid="totalValue">{formatPrice(total)}</div>
    </div>
  );
}
