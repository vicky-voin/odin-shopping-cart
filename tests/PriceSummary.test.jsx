import { describe, it, expect } from "vitest";
import { screen, render } from "@testing-library/react";
import PriceSummary from "../src/components/PriceSummary";
import formatInUSD from "../src/utils/priceConverter";

describe("Price summary tests", () => {
  it("renders 0 for prices if no items passed in", () => {
    const { container } = render(<PriceSummary subtotal={0}></PriceSummary>);
    expect(container).toMatchSnapshot();

    const defaultPrice = formatInUSD(0.0);

    expect(screen.getByTestId("subtotalValue")).toHaveTextContent(defaultPrice);
    expect(screen.getByTestId("shippingValue")).toHaveTextContent(defaultPrice);
    expect(screen.getByTestId("taxValue")).toHaveTextContent(defaultPrice);
    expect(screen.getByTestId("totalValue")).toHaveTextContent(defaultPrice);
  });

  it("renders correct total if subtotal is > 0", () => {
    const testSubtotal = 20;

    const { container } = render(
      <PriceSummary subtotal={testSubtotal}></PriceSummary>
    );

    const expectedShipping = 15;
    const expectedTax = 0.13 * testSubtotal;
    const expectedTotal = testSubtotal + expectedShipping + expectedTax;

    expect(screen.getByTestId("subtotalValue")).toHaveTextContent(
      formatInUSD(testSubtotal)
    );
    expect(screen.getByTestId("shippingValue")).toHaveTextContent(
      formatInUSD(expectedShipping)
    );
    expect(screen.getByTestId("taxValue")).toHaveTextContent(
      formatInUSD(expectedTax)
    );
    expect(screen.getByTestId("totalValue")).toHaveTextContent(
      formatInUSD(expectedTotal)
    );
  });
});
