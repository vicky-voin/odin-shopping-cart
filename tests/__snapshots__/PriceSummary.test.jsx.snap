// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Price summary tests > renders 0 for prices if no items passed in 1`] = `
<div>
  <div>
    <div>
      Subtotal
    </div>
    <div
      data-testid="subtotalValue"
    >
      $0.00
    </div>
    <div>
      Shipping
    </div>
    <div
      data-testid="shippingValue"
    >
      $0.00
    </div>
    <div>
      Tax
    </div>
    <div
      data-testid="taxValue"
    >
      $0.00
    </div>
    <div>
      Order total
    </div>
    <div
      data-testid="totalValue"
    >
      $0.00
    </div>
  </div>
</div>
`;
